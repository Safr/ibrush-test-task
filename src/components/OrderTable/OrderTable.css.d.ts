export const TableWrapper: string;
export const tableWrapper: string;
export const Table: string;
export const table: string;
