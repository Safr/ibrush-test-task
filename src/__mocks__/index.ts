export const orderBook = {
  BTC_USD: {
    ask: ['6891', '0.005', '34.455'],
    ask_amount: '2413543.16939829',
    ask_quantity: '190.96765367',
    ask_top: '6887.20157116',
    bid: ['6849.7394927', '0.75', '5137.30461952'],
    bid_amount: '286086.48310063',
    bid_quantity: '11524.76654836',
    bid_top: '6875',
  },
};

export const order = {
  avg: '6875.86448007',
  buy_price: '6859.227881',
  high: '6969.54201841',
  last_trade: '6864.01',
  low: '6800',
  sell_price: '6873.71642794',
  updated: 1528752429,
  vol: '1112.01492053',
  vol_curr: '7632881.53471028',
};

export const currentOrder = {
  BTC_USD: {
    avg: '6875.86448007',
    buy_price: '6859.227881',
    high: '6969.54201841',
    last_trade: '6864.01',
    low: '6800',
    sell_price: '6873.71642794',
    updated: 1528752429,
    vol: '1112.01492053',
    vol_curr: '7632881.53471028',
  },
};

export const orders = { currentOrder };
export const changedOrders = [''];
export const orderBookInfo = [['5897.8242', '0.003', '17.6934726']];

// export const fetchOrders = () => Promise.resolve({
//   orders: {
//     BTC_USD: {
//       avg: '6875.86448007',
//       buy_price: '6859.227881',
//       high: '6969.54201841',
//       last_trade: '6864.01',
//       low: '6800',
//       sell_price: '6873.71642794',
//       updated: 1528752429,
//       vol: '1112.01492053',
//       vol_curr: '7632881.53471028',
//     },
//   },
// });
