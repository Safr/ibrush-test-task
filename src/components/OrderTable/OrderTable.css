.TableWrapper {
  height: 100px;
  margin-top: 22px;
  overflow-y: scroll;
}

.TableWrapper::-webkit-scrollbar {
  width: 16px;
}

/* Track */
.TableWrapper::-webkit-scrollbar-track {
  box-shadow: inset 0 0 5px #41403e; 
  border-radius: 5px;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background-color: #bbe5f1; 
  border-radius: 10px;
}

/* Handle on hover */
.TableWrapper::-webkit-scrollbar-thumb:hover {
  background-color: #67c5df; 
}

.Table {
  width: 100%;
  border-collapse: collapse;
}

.Table thead th {
  position: sticky;
  top: 0;
  padding-bottom: 12px;
  line-height: 1;
  background-color: #fff;
}

.Table tr:hover {
  background-color: #eaeaea;
}


.Table tbody,
  tr,
  th,
  td {
    padding: 0;
    text-align: left;
    white-space: normal;
}

@media screen and (max-width: 768px) {
  .Table tbody,
    tr,
    th,
    td {
      text-align: center;
  }
}
