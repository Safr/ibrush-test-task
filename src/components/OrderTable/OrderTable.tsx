import React from 'react';
import classes from './OrderTable.css';
import { OrderBookInfo } from '../../types';

interface Props {
  orderBook: OrderBookInfo[];
}

const OrderTable: React.SFC<Props> = ({ orderBook }) => {
  const renderOrderBookInfo = () => {
    let price: string;
    let quantity: string;
    let amount: string;
    return orderBook.map((orderArray, idx) => {
      price = orderArray[0];
      quantity = orderArray[1];
      amount = orderArray[2];
      return (
        <tr key={idx + price}>
          <td>{price}</td>
          <td>{quantity}</td>
          <td>{amount}</td>
        </tr>
      );
    });
  };
  return (
    <div className={classes.TableWrapper}>
      <table className={classes.Table}>
        <thead>
          <tr>
            <th scope="col">Цена</th>
            <th scope="col">Количество</th>
            <th scope="col">Сумма</th>
          </tr>
        </thead>
        <tbody>
          {renderOrderBookInfo()}
        </tbody>
      </table>
    </div>
  );
};

export default OrderTable;
