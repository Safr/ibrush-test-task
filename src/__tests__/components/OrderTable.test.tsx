import React from 'react';
import { shallow } from 'enzyme';
import OrderTable from '../../components/OrderTable/OrderTable';
import { orderBookInfo } from '../../__mocks__';

describe('OrderTable Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<OrderTable orderBook={orderBookInfo} />);
    expect(wrapper).toMatchSnapshot();
  });
});
