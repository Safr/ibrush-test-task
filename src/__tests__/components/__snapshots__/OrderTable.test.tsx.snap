// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderTable Component should render correctly 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th
          scope="col"
        >
          Цена
        </th>
        <th
          scope="col"
        >
          Количество
        </th>
        <th
          scope="col"
        >
          Сумма
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="05897.8242"
      >
        <td>
          5897.8242
        </td>
        <td>
          0.003
        </td>
        <td>
          17.6934726
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
